### PR TITLE
Feature/#96 add arbitrary polygon support

### PR DESCRIPTION
**Pull request by hixi**
Part of #96 

Adds support for an clipping area.

Reviewed by
- [x] @das-g

### CODE REVIEW

File: Dockerfile.rest_api [hunk 1]
```diff
@@ -15,7 +18,7 @@ WORKDIR $HOME
 
 ADD requirements.txt $HOME/
 
-RUN pip install honcho
+RUN pip3 install honcho
```
Comment (hixi) [inline Dockerfile.rest_api]: switched to python3 so we don't need to install another python version (besides the standard python3).

File: Dockerfile.rest_api [hunk 2]
```diff
@@ -6,7 +8,8 @@ RUN DEBIAN_FRONTEND=noninteractive apt-get update && \
     DEBIAN_FRONTEND=noninteractive apt-get install -y\
 \
     libgeos-dev \
-    libgeos++-dev
+    libgeos++-dev \
```
Comment (hixi) [inline Dockerfile.rest_api]: seems top be a dependency for the newer version of `djangorestframework-gis`, therefore included.

File: conversion_service/clipping_geometry/serializers.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    def validate_clipping_multi_polygon(self, clipping_multi_polygon):
+        if type(clipping_multi_polygon) != geos.MultiPolygon:
+            raise serializers.ValidationError(
+                _(
+                    "The received geometry is not a MultiPolygon. Received type of {other_type}.".format(
+                        other_type=clipping_multi_polygon.__class__.__name__
+                    )
+                )
+            )
+        if clipping_multi_polygon.num_coords <= 0:
```
Comment (hixi) [inline conversion_service/clipping_geometry/serializers.py]: if it has not more than one coordinate, it's not valid, but the geoJSON to python conversion doesn't catch this and the error is raised on database level (integrity error), which doesn't leave much information, why it happened.

File: Dockerfile.rest_api [hunk 4]
```diff
@@ -6,7 +8,8 @@ RUN DEBIAN_FRONTEND=noninteractive apt-get update && \
...
+    python3-pip
```
Comment (hixi) [inline Dockerfile.rest_api]: other image, apparently doesn't include python3-pip

File: conversion_service/clipping_geometry/serializers.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+                    "The received geometry is not a MultiPolygon. Received type of {other_type}.".format(
+                        other_type=clipping_multi_polygon.__class__.__name__
+                    )
+                )
+            )
+        # prevent obscure IntegrityError in the database:
+        # occurs for example when the geoJSON seems valid, but contains non-coordinate string(s).
+        if clipping_multi_polygon.num_coords <= 0:
+            raise serializers.ValidationError(
+                _("Invalid coordinates: Expecting at least one coordinate set, received none.")
```
Comment (das-g) [inline conversion_service/clipping_geometry/serializers.py]: 'set' might induce the wrong association with the concept of a mathematical set (an unordered set). How about 'coordinate tuple', or &ndash; as we're only dealing with 2D &ndash; 'coordinate pair'?

Comment (hixi) [inline conversion_service/clipping_geometry/serializers.py]: good idea :-)

**Pull request merged**